test/unit/StakingNFT/stakingNFT.js

```diff
@@ -53,9 +66,101 @@ describe('StakingNFT', function () {
+    const [nonOwner, owner] = this.accounts.members;
+    await stakingNFT.connect(this.stakingPoolSigner).mint(this.poolId, owner.address);
+    await expect(cover.operatorTransferFrom(nonOwner.address, owner.address, 0)).to.be.revertedWith('WRONG_FROM');
+  });
+
+  it('should revert if trying to operatorTransfer a token to a zero address - ZERO_ADDRESS', async function () {
+    const { stakingNFT, cover } = this;
+    const [owner] = this.accounts.members;
+    await stakingNFT.connect(this.stakingPoolSigner).mint(this.poolId, owner.address);
+    await expect(cover.operatorTransferFrom(owner.address, ethers.constants.AddressZero, 0)).to.be.revertedWith(
+      'INVALID_RECIPIENT',
+    );
+  });
+
+  it('should revert if trying to transferFrom a token from a non-owner - WRONG_FROM', async function () {
+    const { stakingNFT } = this;
+    const [nonOwner, owner] = this.accounts.members;
+    await stakingNFT.connect(this.stakingPoolSigner).mint(this.poolId, owner.address);
+    await expect(stakingNFT.transferFrom(nonOwner.address, owner.address, 0)).to.be.revertedWith('WRONG_FROM');
+  });
+
+  it('should revert if trying to transferFrom a token to a zero address - ZERO_ADDRESS', async function () {
+    const { stakingNFT } = this;
+    const [owner] = this.accounts.members;
+    await stakingNFT.connect(this.stakingPoolSigner).mint(this.poolId, owner.address);
+    await expect(stakingNFT.transferFrom(owner.address, ethers.constants.AddressZero, 0)).to.be.revertedWith(
+      'INVALID_RECIPIENT',
+    );
+  });
+
+  it('should revert if not approved to transferFrom a token - NOT_AUTHORIZED', async function () {
+    const { stakingNFT } = this;
+    const [owner, nonOwner] = this.accounts.members;
+    await stakingNFT.connect(this.stakingPoolSigner).mint(this.poolId, owner.address);
+    await expect(stakingNFT.connect(nonOwner).transferFrom(owner.address, nonOwner.address, 0)).to.be.revertedWith(
+      'NOT_AUTHORIZED',
+    );
+  });
+
+  it('should transferFrom a token', async function () {
+    const { stakingNFT } = this;
+    const [owner, nonOwner] = this.accounts.members;
+    await stakingNFT.connect(this.stakingPoolSigner).mint(this.poolId, owner.address);
+    await stakingNFT.connect(owner).approve(nonOwner.address, 0);
+    await stakingNFT.connect(nonOwner).transferFrom(owner.address, nonOwner.address, 0);
+    expect(await stakingNFT.ownerOf(0)).to.be.equal(nonOwner.address);
+  });
+
+  // TODO: why is safeTransferFrom() not found?
+  it.skip('should fail to safeTransfer to a contract that does not implement onERC721Received', async function () {
+    const { stakingNFT, cover } = this;
+    const [owner] = this.accounts.members;
+    await stakingNFT.connect(this.stakingPoolSigner).mint(this.poolId, owner.address);
+    await expect(
+      stakingNFT.connect(owner).safeTransferFrom(owner.address, cover.address, BigNumber.from(0), toBytes('cafe')),
+    ).to.be.revertedWith('NOT_ERC721_RECEIVER');
+  });
```

Comment:
`safeTransferFrom()` is not found because there are two implementations of it. It is needed to provide the full signature of the method to use. In this case:
```js
await expect(
  stakingNFT
    .connect(owner)
    ['safeTransferFrom(address,address,uint256,bytes)'](
      owner.address,
      cover.address,
      BigNumber.from(0),
      toBytes('cafe'),
    ),
).to.be.revertedWith('NOT_ERC721_RECEIVER');
```

I tried it, and it works. There is one issue with the expected revert message  `NOT_ERC721_RECEIVER`. It is reverting without a reason. 

The problem is that the implementation of solmate does the following:
```solidity
require(
  to.code.length == 0 ||
  ERC721TokenReceiver(to).onERC721Received(msg.sender, from, id, data)
    == ERC721TokenReceiver.onERC721Received.selector,
  "UNSAFE_RECIPIENT"
);
```

As the contract does not implement `onERC721Received()`, the call `ERC721TokenReceiver(to).onERC721Received(msg.sender, from, id, data)` reverts without a reason before comparing the returned value. That revert message will only be thrown if a contract implements the method but does not return the correct value.

This is handled differently in [OpenZeppelin implementation](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/91e8d0ba3c3beb8a1db31310d8599664e48639ef/contracts/token/ERC721/ERC721.sol#L406-L416).

I'm not sure if it is worth updating the contract implementation as the result is the same. As an alternative, we can just add a comment in the tests explaining why they are reverting in that way. Thoughts? @kyledewy @shark0der

Reply:
Oh nice! I totally forgot you could call functions like that.
I would be fine to leave it a silent revert, since it's the same result.